ambry-store/src/main/java/com.github.ambry.store/CompactionLog.java

```diff
@@ -268,19 +277,23 @@ private void flush() {
    */
   private static class CycleLog {
     private static final int TIMESTAMP_SIZE = 8;
+    private static final int STORE_TOKEN_PRESENT_FLAG_SIZE = 1;
+
+    private static final byte STORE_TOKEN_PRESENT = (byte) 1;
+    private static final byte STORE_TOKEN_ABSENT = (byte) 0;
 
     // details about the cycle
     final CompactionDetails compactionDetails;
     // start time of the copy phase
-    long copyStartTime = -1;
-    // start time of the switch phase
-    long switchStartTime = -1;
+    long copyStartTime = UNINITIALIZED_TIMESTAMP;
+    // start time of the commit phase
+    long commitStartTime = UNINITIALIZED_TIMESTAMP;
     // start time of the cleanup phase
-    long cleanupStartTime = -1;
+    long cleanupStartTime = UNINITIALIZED_TIMESTAMP;
     // end time of the current cycle
-    long cycleEndTime = -1;
+    long cycleEndTime = UNINITIALIZED_TIMESTAMP;
     // point until which copying is complete
-    StoreFindToken safeToken = new StoreFindToken();
+    StoreFindToken safeToken = null;
```

Comment:
I would like to differentiate from the uninited `StoreFindToken` which has the connotation of "start from the beginning of the log" and this case where it is "start from wherever you need to".